### PR TITLE
feat(emails): improve tests for email notifications

### LINKED ISSUE #2795: [testing] Improve testing of notification email content
**Is your feature request related to a problem? Please describe.**
Did you know the platform sends notification emails to users?! Pretty cool. You can find the code in this [directory](https://github.com/ONEARMY/community-platform/tree/evakill/moderation-emails/functions/src/emailNotifications) in the functions workspace. The [readme](https://github.com/ONEARMY/community-platform/blob/evakill/moderation-emails/functions/src/emailNotifications/README.md) has more information about the email process.

**Describe the solution you'd like**
The file [createNotificationEmails](https://github.com/ONEARMY/community-platform/blob/evakill/moderation-emails/functions/src/emailNotifications/createNotificationEmails.ts) file handles the logic of generating the email HTML and inserting the email into the database. In the test for this file, we generate a snapshot of the HTML. This is a good check to ensure the emails we're sending match our design specs. However, snapshots of a big HTML blob are not very readable, and we really want to ensure we're sending the correct content in our emails to users. 

To improve testing, please create some expect statements that check the HTML for dynamically generated content. For example, it would be important to check that a we're sending the right message, i.e. "New comment on your how-to by User 1", and the HTML contains the correct link, i.e. https://community.preciousplastic.com/test. 

For an example, check out this [commit](https://github.com/ONEARMY/community-platform/pull/2792/commits/8cba6679f27577ffc1b66bd96c9aa9578b193989), which does the same for moderation emails. 

You run these tests locally with the command `yarn workspace functions test:watch`

### PR DESCRIPTION
**Pull request by pizzaisdavid**
PR Checklist

- [ ] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Developer experience (improves developer workflows for contributing to the project)

## Description

_What this PR does_

Adds some specific asserts for user facing content.

## Git Issues

Closes #2795

## Screenshots/Videos

_If useful, provide screenshot or capture to highlight main changes_

---

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of a monthly dev call (first Monday of the month, open to all!).

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on Discord in the [Community Platform `development` channel](https://discord.com/channels/586676777334865928/938781727017558018).

### CODE REVIEW

**Comment by AlfonsoGhislieri** [comment]
Thanks for raising this @pizzaisdavid, however we are already checking the html in the test against the snapshot file with:

[`expect(html).toMatchSnapshot()` 
](https://github.com/ONEARMY/community-platform/blob/185dff79546a92ff00f8b511bc0126df2f8a3b64/functions/src/emailNotifications/createNotificationEmails.spec.ts#L211)

Closing this as it adds duplicate assertions.

**Comment by pizzaisdavid** [comment]
@AlfonsoGhislieri please then also close the corresponding ticket (https://github.com/ONEARMY/community-platform/issues/2795) or if I've done something different than the requirements of the ticket.

**Comment by AlfonsoGhislieri** [comment]
No no you did the correct implementation, I see the point @evakill raised about the readability, but to me it doesn’t seem like these assertions actually add much, unless the idea here is to detach the checking of the structure of the email and the actual content.

I’ll reopen up and leave it to them as they raised the initial issue.

**Comment by evakill** [comment]
@pizzaisdavid thanks for picking this up! 

and thanks for the review @AlfonsoGhislieri - agree that the code is in a bit of a transitional state, but i can give more context:

so, these notification digest emails were the first to be implemented. they are generated using some frankenstein-html, which you can look at in the `getNotificationEmailHtml` function. since then we have put more work into "transactional" emails, which are triggered on an event - i.e. when a user's how-to is approved, when a user is given a "verified" badge. these transactional emails use handlebars for templating and a development vite server to preview the email format. this has been a super helpful improvement in cleaning up the code and creating an easier development process. 

we're moving away from snapshots, which are less readable, to a combination of this development preview system and targeted unit tests. the notification emails are the last to _not_ use the handlebars system. they're a little more complex, so will take a bit of refactoring to do this move, but it's something that i'd like to get done soon.

TLDR; this issue was inspired in part by an upcoming refactor and in part by a regression that we didn't catch in review (proving snapshots were not doing their job). goal remains the same:
> However, snapshots of a big HTML blob are not very readable, and we really want to ensure we're sending the correct content in our emails to users.

going to resuming reviewing... hope this helps

File: functions/src/emailNotifications/createNotificationEmails.spec.ts [hunk 1]
```diff
@@ -209,6 +209,19 @@ describe('create email test', () => {
         to,
       } = doc.data()
       expect(html).toMatchSnapshot()
+      expect(html).toContain(
+        `<p><a href='https://community.preciousplastic.com/u/user_2'>User 2</a> found your <a href='https://community.preciousplastic.com/test'>how-to</a> useful</p>`,
+      )
+      expect(html).toContain(
```
Comment (evakill) [inline functions/src/emailNotifications/createNotificationEmails.spec.ts:215]: can we extract the "content" from the HTML? it will make the tests less fragile. i.e. instead of testing 
```
      expect(html).toContain(
        `<p>New comment on your <a href='https://community.preciousplastic.com/test'>research</a> by <a href='https://community.preciousplastic.com/u/user_2'>User 2</a></p>`,
      )
```
could we test 
```
      expect(html).toContain('New comment')
      expect(html).toContain('https://community.preciousplastic.com/test')
      expect(html).toContain('https://community.preciousplastic.com/u/user_2')
```

**Review by evakill — CHANGES_REQUESTED** [review]

**Comment by evakill** [comment]
closing due to inactivity
